src/c/service.c

```diff
@@ -999,10 +1027,15 @@ void devsdk_service_start (devsdk_service_t *svc, iot_data_t *driverdfls, devsdk
   }
   else
   {
-    edgex_device_parseClients (svc->logger, iot_data_string_map_get (config_file, "Clients"), &svc->config.endpoints);
+    edgex_device_parseClients (svc->logger, iot_data_string_map_get (deviceservices_config, "Clients"), &svc->config.endpoints);
   }
 
   iot_data_free (config_file);
+  if (common_config_file)
```

Comment:
`iot_data_free()` follows the `free()` convention of allowing a NULL parameter, so you don't need to protect it with an if

Reply:
updated